packages/react-native/ReactAndroid/src/main/java/com/facebook/react/animated/FrameBasedAnimationDriver.java

```diff
@@ -70,7 +73,14 @@ public void runAnimationStep(long frameTimeNanos) {
     long timeFromStartMillis = (frameTimeNanos - mStartFrameTimeNanos) / 1000000;
     int frameIndex = (int) Math.round(timeFromStartMillis / FRAME_TIME_MILLIS);
     if (frameIndex < 0) {
-      throw new IllegalStateException("Calculated frame index should never be lower than 0");
+      String message = "Calculated frame index should never be lower than 0. Called with frameTimeNanos " +
+        frameTimeNanos + " and mStartFrameTimeNanos " + mStartFrameTimeNanos;
+      if (ReactBuildConfig.DEBUG) {
+        throw new IllegalStateException(message);
+      } else {
+        FLog.w(ReactConstants.TAG, message);
```

Comment:
I'm unsure how this fixes given that `frameIndex` will then be negative by then and it's used below to index access an array?

Reply:
It just returns early to make sure no other effects of this function run (see line 82):

https://github.com/facebook/react-native/pull/37487/files#diff-16ce46a8c2c1aa9e045a19ea56f11d5043172f9086bc910adb15585bf08a830bR82

Does that work? I think it should be fairly safe since we basically pretend that this animation shouldn't have its frame change, which should be a no-op.

Reply:
> It just returns early

You're so right :) I missed that line